go/cmd/vtctldclient/command/shard_routing_rules.go

```diff
@@ -87,7 +87,7 @@ func commandApplyShardRoutingRules(cmd *cobra.Command, args []string) error {
 	}
 
 	srr := &vschemapb.ShardRoutingRules{}
-	if err := json2.Unmarshal(rulesBytes, &srr); err != nil {
+	if err := json2.UnmarshalPB(rulesBytes, srr); err != nil {
```

Comment:
Seems like there may be other places where it is worth changing the usage to `UnmarshalPB`:
```
❯ git grep json2\.Unmarshal | grep -vE 'test|wrangler|vtctl.go'
go/cmd/vtctldclient/command/keyspace_routing_rules.go:	if err := json2.Unmarshal(rulesBytes, &krr); err != nil {
go/cmd/vtctldclient/command/routing_rules.go:	if err := json2.Unmarshal(rulesBytes, &rr); err != nil {
go/cmd/vtctldclient/command/shard_routing_rules.go:	if err := json2.Unmarshal(rulesBytes, &srr); err != nil {
go/cmd/vtctldclient/command/vschemas.go:		err = json2.Unmarshal(schema, &vs)
go/vt/tableacl/tableacl.go:		if jsonErr := json2.Unmarshal(data, config); jsonErr != nil {
go/vt/vtctl/workflow/traffic_switcher.go:		if err := json2.Unmarshal([]byte(wrap), ks); err != nil {
go/vt/vtexplain/vtexplain_vtgate.go:	err := json2.Unmarshal([]byte(wrappedStr), &srvVSchema)
go/vt/vtexplain/vtexplain_vtgate.go:	err := json2.Unmarshal([]byte(ksShardMapStr), &ksShardMap)
go/vt/vtgate/planbuilder/fuzz.go:	err = json2.Unmarshal(data, formal)
go/vt/vtgate/vindexes/vschema.go:	err = json2.Unmarshal(data, formal)
go/vt/vtgate/vindexes/vschema.go:	err = json2.Unmarshal(data, formal)
```

We don't have to change them all here but would be nice to IMO so that we don't potentially leave current or future latent cases of the same kind around.

Reply:
There are other such places across the code base. We should clean them all up separately.